models/issue_tracked_time.go

```diff
@@ -131,3 +225,106 @@ func TotalTimes(options FindTrackedTimesOptions) (map[*User]string, error) {
 	}
 	return totalTimes, nil
 }
+
+// DeleteIssueUserTimes deletes times for issue
+func DeleteIssueUserTimes(issue *Issue, user *User) error {
+	sess := x.NewSession()
+	defer sess.Close()
+
+	if err := sess.Begin(); err != nil {
+		return err
+	}
+
+	opts := FindTrackedTimesOptions{
+		IssueID: issue.ID,
+		UserID:  user.ID,
+	}
+
+	removedTime, err := deleteTimes(sess, opts)
+	if err != nil {
+		return err
+	}
+	if removedTime == 0 {
+		return ErrNotExist{}
+	}
+
+	if err := issue.loadRepo(sess); err != nil {
+		return err
+	}
+	if _, err := createComment(sess, &CreateCommentOptions{
+		Issue:   issue,
+		Repo:    issue.Repo,
+		Doer:    user,
+		Content: "- " + SecToTime(removedTime),
+		Type:    CommentTypeDeleteTimeManual,
+	}); err != nil {
+		return err
+	}
+
+	return sess.Commit()
+}
+
+// DeleteTime delete a specific Time
+func DeleteTime(t *TrackedTime) error {
+	sess := x.NewSession()
+	defer sess.Close()
+
+	if err := sess.Begin(); err != nil {
+		return err
+	}
+
+	if err := deleteTime(sess, t); err != nil {
+		return err
+	}
+
+	if _, err := createComment(sess, &CreateCommentOptions{
+		Issue:   t.Issue,
+		Repo:    t.Issue.Repo,
+		Doer:    t.User,
+		Content: "- " + SecToTime(t.Time),
+		Type:    CommentTypeDeleteTimeManual,
+	}); err != nil {
+		return err
+	}
+
+	return sess.Commit()
+}
+
+func deleteTimes(e Engine, opts FindTrackedTimesOptions) (removedTime int64, err error) {
+	sess := x.NewSession()
+	defer sess.Clone()
+
+	removedTime, err = getTrackedSeconds(sess, opts)
+	if err != nil {
+		return 0, err
+	}
+
+	_, err = opts.ToSession(sess).SetExpr("deleted", true).Update(&TrackedTime{})
+	if err != nil {
+		return 0, err
+	}
+	return removedTime, sess.Commit()
```

Comment:
```suggestion
	return removedTime, err
```

Reply:
think https://github.com/go-gitea/gitea/pull/9200/files#r361540946 is more elegant